tests/unit/kernel/MaternKernel_unittest.cc

```diff
@@ -0,0 +1,142 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Gil Hoben
+ */
+#include <gtest/gtest.h>
+#include <shogun/kernel/MaternKernel.h>
+#include <shogun/lib/SGMatrix.h>
+#include <shogun/util/zip_iterator.h>
+
+using namespace shogun;
+
+TEST(MaternKernelTest_1_2, test_kernel_matrix)
+{
+	SGMatrix<float64_t> matrix{{0, 1, 2}, {3, 4, 5}};
+	SGMatrix<float64_t> expected_kernel_matrix{{1., 0.07441661},
+											   {0.07441661, 1.}};
+
+	auto features = std::make_shared<DenseFeatures<float64_t>>(matrix);
+
+	auto kernel = std::make_shared<MaternKernel>(features, features, 10, 2.0, 0.5);
+	SGMatrix<float64_t> computed_kernel_matrix = kernel->get_kernel_matrix();
+
+	for (const auto& [result, reference]: zip_iterator(computed_kernel_matrix, expected_kernel_matrix))
+		EXPECT_NEAR(result, reference, 1E-7);		
+}
+
+TEST(MaternKernelTest_3_2, test_kernel_matrix)
+{
+	SGMatrix<float64_t> matrix{{0, 1, 2}, {3, 4, 5}};
+	SGMatrix<float64_t> expected_kernel_matrix{{1., 0.06109951},
+											   {0.06109951, 1.}};
+
+	auto features = std::make_shared<DenseFeatures<float64_t>>(matrix);
+
+	auto kernel = std::make_shared<MaternKernel>(features, features, 10, 2.0, 1.5);
+	SGMatrix<float64_t> computed_kernel_matrix = kernel->get_kernel_matrix();
+
+	for (const auto& [result, reference]: zip_iterator(computed_kernel_matrix, expected_kernel_matrix))
+		EXPECT_NEAR(result, reference, 1E-7);		
+}
+
+TEST(MaternKernelTest_5_2, test_kernel_matrix)
+{
+	SGMatrix<float64_t> matrix{{0, 1, 2}, {3, 4, 5}};
+	SGMatrix<float64_t> expected_kernel_matrix{{1., 0.05416044},
+											   {0.05416044, 1.}};
+
+	auto features = std::make_shared<DenseFeatures<float64_t>>(matrix);
+
+	auto kernel = std::make_shared<MaternKernel>(features, features, 10, 2.0, 2.5);
+	SGMatrix<float64_t> computed_kernel_matrix = kernel->get_kernel_matrix();
+
+	for (const auto& [result, reference]: zip_iterator(computed_kernel_matrix, expected_kernel_matrix))
+		EXPECT_NEAR(result, reference, 1E-7);		
+}
+
+TEST(MaternKernelTest_1_2, test_derivative_width)
+{
+	// gradient computed with JAX: https://gist.github.com/gf712/b10ecefd2379e8492c9802ac8cb329c9
```

Comment:
@karlnapf getting the same results as JAX but different from sklearn..

Reply:
ha! Is the kernel maybe parametrized differently?

Reply:
I’m not sure.. the kernel matrix is the same, it’s just the derivative that is different

Reply:
Could you actually check the gist please? I think it’s correct.. kernel wrt to width right?

Reply:
It seems like the sklearn gradient is exactly double of what I’m getting..

Reply:
So sklearn overestimates the gradient by a factor of `width`. I tested it numerically, and the JAX/WolframAlpha gradients are correct. I wonder why.. I suspect it is related to this distance calculation
https://github.com/scikit-learn/scikit-learn/blob/95d4f0841d57e8b5f6b2a570312e9d832e69debc/sklearn/gaussian_process/kernels.py#L1437-L1438

Reply:
maybe it has to do with the fact how they use the gradients downstream

Reply:
maybe, maybe it also doesn't matter too much as long as they are in correct within the same factor.. Anyway, this should be correct